clause/expression.go

```diff
@@ -124,6 +125,14 @@ func (expr NamedExpr) Build(builder Builder) {
 	name := make([]byte, 0, 10)
 
 	for _, v := range []byte(expr.SQL) {
+		if quotationMarks == v && (v == '"' || v == '\'') {
+			quotationMarks = 0
+		} else if quotationMarks == 0 && (v == '"' || v == '\'') {
+			quotationMarks = v
+		} else if quotationMarks == '"' || quotationMarks == '\'' {
+			builder.WriteByte(v)
+			continue
+		}
 		if v == '@' && !inName {
```

Comment:
I think it would be better to add a check here to ensure that the previous character is not a " or '

Reply:
The logic of this code is
**The second if:**
When we first encounter '' or '', quotationMarks will be marked as' 'or' 'accordingly`
**The first if:**
When we have already been marked and encountered that mark (end mark) again, we will clear the mark
**The third if:**
When we have been marked and have not encountered the end mark, we will output it as it is

I didn't understand where you wanted to add a judgment